ngx_http_auth_ldap_module.c

```diff
@@ -1334,18 +1345,31 @@ ngx_http_auth_ldap_ssl_handshake_handler(ngx_connection_t *conn, ngx_flag_t vali
           long chain_verified = SSL_get_verify_result(conn->ssl->connection);
 
           int addr_verified;
-          char *hostname = c->server->ludpp->lud_host;
-          addr_verified = X509_check_host(cert, hostname, 0, 0, 0);
-
-          if (!addr_verified) { // domain not in cert? try IP
-            size_t len; // get IP length
-            if (conn->sockaddr->sa_family == 4) len = 4;
-            else if (conn->sockaddr->sa_family == 6) len = 16;
-            else { // very unlikely indeed
-              ngx_http_auth_ldap_close_connection(c);
-              return;
+          if (c->server->ssl_check_cert == SSL_CERT_VERIFY_CHAIN) {
+            // chain_verified is enough, not requiring full name/IP verification
+            addr_verified = 1;
+
+          } else {
+            // verify hostname/IP
+            char *hostname = c->server->ludpp->lud_host;
+            addr_verified = X509_check_host(cert, hostname, 0, 0, 0);
+
+            if (!addr_verified) { // domain not in cert? try IP
+              size_t len; // get IP length
+
+              struct sockaddr *conn_sockaddr = NULL;
```

Comment:
This next few lines of code was what was causing the segfault in kvspb/nginx-auth-ldap/#236. `conn->sockaddr` was always `NULL` for me. So check a few places that (I *think*) should have the value we need and use one of them, if set. If there's anywhere my logic might be wrong, it would be here, in assuming the values of these `sockaddr` structs represent the same thing. But I think it's right.